### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by danispavel**
User profile with editing  and minimization changes

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -34,15 +34,17 @@ class App extends Component {
...
+      this.listener = this.props.firebase
+        .auth()
+        .onAuthStateChanged(authUser => {
+          authUser
+            ? this.setState({
+                authUser
+              })
+            : this.setState({
+                authUser: null
+              });
```
Comment (SaenkoM) [inline src/App.js]:
```
this.setState({
  authUser: authUser || null,
})
```

File: src/js/components/Admin.jsx [hunk 2]
```diff
@@ -1,76 +1,82 @@
...
-		this.state = {
-			loading: false,
-			users: []
-		};
-	}
+    this.state = {
+      loading: false,
+      users: []
+    };
+  }
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Admin.jsx]: Chenged

File: src/js/components/Admin.jsx [hunk 3]
```diff
@@ -1,76 +1,82 @@
...
 
-	componentDidMount() {
-		this.setState({ loading: true });
+  componentDidMount() {
+    this.setState({ loading: true });
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Admin.jsx]: Changed

File: src/js/components/Auth/requireAuth.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+export default ChildComponent => {
+  const ComposedComponent = props => {
+    useEffect(() => {
+      if (props.auth.isLoaded && props.auth.isEmpty)
```
Comment (SaenkoM) [inline src/js/components/Auth/requireAuth.jsx]: Please use object destructuring for props whenever you can.

File: src/js/components/Auth/index.jsx [hunk 5]
```diff
@@ -1,5 +1,6 @@
-import AuthUserContext from './context';
-import withAuthentication from './withAuthentication';
-import withAuthorization from './withAuthorization';
+import AuthUserContext from "./context";
+import withAuthentication from "./withAuthentication";
+import withAuthorization from "./withAuthorization";
+import requireAuth from "./requireAuth";
 
-export { AuthUserContext, withAuthentication, withAuthorization };
+export { AuthUserContext, withAuthentication, withAuthorization, requireAuth };
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Auth/index.jsx]: I have to add export default?

File: src/js/components/Admin.jsx [hunk 6]
```diff
@@ -1,76 +1,82 @@
...
+    const { users, loading } = this.state;
 
-		return (
-			<AuthUserContext.Consumer>
-				{authUser => (
-					<div>
-						<h1>Admin</h1>
-						{loading && <div>Loading ...</div>}
+    return (
+      <AuthUserContext.Consumer>
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Admin.jsx]: Check

File: src/js/components/Auth/requireAuth.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        return props.history.push("/login");
+    }, [props.auth, props.history]);
+
+    return <ChildComponent {...props} />;
+  };
+
+  function mapStateToProps(state) {
```
Comment (SaenkoM) [inline src/js/components/Auth/requireAuth.jsx]: Prefer const declaration, can be like:
```
const mapStateToProps = state => ({
  auth: state.firebaseReducer.auth,
})
```

File: src/js/components/Auth/requireAuth.jsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+export default ChildComponent => {
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Auth/requireAuth.jsx]: Changed

File: src/js/components/NavBar.jsx [hunk 9]
```diff
@@ -1,169 +1,167 @@
...
-		this.handleClickOutside = this.handleClickOutside.bind(this);
-	}
+  constructor(props) {
+    super(props);
+    this.state = {
+      authUser: null,
+      isActive: false
+    };
+    this.setWrapperRef = this.setWrapperRef.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
```
Comment (SaenkoM) [inline src/js/components/NavBar.jsx]: Please refactor as arrow functions and remove binds, then state declaration can be extracted from constructor and constructor removed.

File: src/js/components/Auth/requireAuth.jsx (hunk 4 above)
Comment (danispavel) [reply src/js/components/Auth/requireAuth.jsx]: Changed

File: src/js/components/Auth/requireAuth.jsx (hunk 7 above)
Comment (danispavel) [reply src/js/components/Auth/requireAuth.jsx]: Changed

File: src/js/components/NavBar.jsx (hunk 9 above)
Comment (danispavel) [reply src/js/components/NavBar.jsx]: Changed

File: src/js/components/NavBar.jsx [hunk 10]
```diff
@@ -1,169 +1,167 @@
...
-										Account
-									</Link>
-								)}
+            <div
+              id="navbarBasicExample"
+              className={isActive ? "navbar-menu is-active" : "navbar-menu"}
+            >
+              <div className="navbar-start">
+                {authUser && (
+                  <a
```
Comment (SaenkoM) [inline src/js/components/NavBar.jsx]: Prefer `Link` component

File: src/js/components/Profile/Profile.jsx [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+  componentWillUnmount() {
+    if (this.props.firebase) {
+      this.listener();
+    }
+  }
+
+  render() {
+    const { bio, full_name, profile_picture } = this.props.profile;
+    const { history } = this.props;
```
Comment (SaenkoM) [inline src/js/components/Profile/Profile.jsx]:
```
const { history, profile } = this.props;
const { bio, full_name, profile_picture } = profile;
```
or:
```
const { history, profile:  { bio, full_name, profile_picture } } = this.props;
```

File: src/js/components/ProfileEditing/ProfileEditing.jsx [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  state = {
+    authUser: null
+  };
+
+  removeHandler() {
+    this.setState({
+      profile_picture:
+        "https://firebasestorage.googleapis.com/v0/b/moxiegirloauthin-1559407531552.appspot.com/o/default_avatar.png?alt=media&token=00ce67f4-6fe7-481d-a49f-bde73f5fbe60"
```
Comment (SaenkoM) [inline src/js/components/ProfileEditing/ProfileEditing.jsx]: Should be a constant

File: src/js/containers/LandingPage/LandingPage.jsx [hunk 13]
```diff
@@ -38,19 +36,37 @@ class LandingPage extends Component {
...
         .then(res => {
-          const { id, username, full_name, bio } = res.data.data;
+          const { id, full_name, bio, profile_picture } = res.data.data;
+
           if (firebase) {
             firebase
-              .doSignInWithEmailAndPassword(`${id}@gmail.com`, id)
+              .auth()
+              .signInWithEmailAndPassword(`${id}@gmail.com`, id)
+              // .then(res => {
```
Comment (SaenkoM) [inline src/js/containers/LandingPage/LandingPage.jsx]: Commented code should be a rare exception

File: src/js/components/NavBar.jsx (hunk 10 above)
Comment (danispavel) [reply src/js/components/NavBar.jsx]: Changed

File: src/js/redux/reducers/profile.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  full_name: "Moxie_Girl",
+  profile_picture:
+    "https://firebasestorage.googleapis.com/v0/b/moxiegirloauthin-1559407531552.appspot.com/o/default_avatar.png?alt=media&token=00ce67f4-6fe7-481d-a49f-bde73f5fbe60",
+  profilePictureRefPath: null
+};
+const reducer = handleActions(
+  {
+    [setProfileData]: (state, action) => {
+      const { ...rest } = action.payload;
+      return { ...state, ...rest };
```
Comment (SaenkoM) [inline src/js/redux/reducers/profile.js]: That would work same
```
return {
  ...state,
  ...action.payload,
}
```
To handle correct spread you could write `(state, { payload = {} })`

File: src/js/components/Profile/Profile.jsx (hunk 11 above)
Comment (danispavel) [reply src/js/components/Profile/Profile.jsx]: Changed

File: src/js/components/Profile/container.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import { connect } from "react-redux";
+import { setProfileData } from "../../redux/actions/profile";
+
+function mapStateToProps(state) {
```
[the remark this answers is not in the document]
Comment (danispavel) [reply src/js/components/Profile/container.js]: Changed

File: src/js/containers/LandingPage/LandingPage.jsx (hunk 13 above)
Comment (danispavel) [reply src/js/containers/LandingPage/LandingPage.jsx]: Deleted

File: src/js/components/Auth/index.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (SaenkoM) [reply src/js/components/Auth/index.jsx]: Sure, just check that it will work right

**Pull request merged**